net/instaweb/rewriter/rewrite_context.cc

```diff
@@ -1996,13 +1998,28 @@ void RewriteContext::Harvest() {
 void RewriteContext::Render() {
 }
 
+bool RewriteContext::AreOutputsAllowedByCsp(CspDirective role) const {
+  if (Driver()->content_security_policy().empty()) {
+   return true;
+  }
+
+  for (const OutputResourcePtr& o : outputs_) {
+    if (o.get() != nullptr && o->has_hash() && o->has_url() &&
```

Comment:
Why is it necessary to consider `o->has_hash()` here?

Reply:
Did your run the tests with a DCHECK?